app/services/schools/dfe_sign_in_api/client.rb

```diff
@@ -0,0 +1,37 @@
+module Schools
+  module DFESignInAPI
+    class Client
+      def self.enabled?
+        [
+          ENV['DFE_SIGNIN_API_CLIENT'],
+          ENV['DFE_SIGNIN_API_SECRET']
+        ].map(&:presence).all?
+      end
+      delegate :enabled?, to: :class
+
+    private
+
+      def response
+        return [] unless enabled?
+
+        resp = Faraday.new.get(endpoint) do |req|
+          req.headers['Authorization'] = "bearer #{token}"
+          req.headers['Content-Type']  = 'application/json'
+        end
+
+        JSON.parse(resp.body)
```

Comment:
This will need to handle bad responses - eg check response code, and returned content type - before parsing it out as JSON?

Reply:
Even if it just raises an exception with some useful information such as the response code - that would be enough to call out the source of a problem

Reply:
Exception handling has been added using the `Faraday::Response::RaiseError`